tests/php/Authenticator/LoginHandlerTest.php

```diff
@@ -0,0 +1,192 @@
+        ]], $response['registrationDetails']);
+    }
+
+    public function testMFASchemaEndpointShowsRegisteredMethodsIfSetUp()
+    {
+        // "Simon" is security conscious - he uses the cutting edge MFA methods
+        $member = $this->objFromFixture(Member::class, 'simon');
+        $this->scaffoldPartialLogin($member);
+
+        $result = $this->get('Security/login/default/mfa/schema');
+
+        $response = json_decode($result->getBody(), true);
+
+        $this->assertArrayHasKey('registeredMethods', $response);
+        $this->assertArrayHasKey('registrationDetails', $response);
+        $this->assertArrayHasKey('defaultMethod', $response);
+
+        $this->assertCount(1, $response['registeredMethods']);
+        $this->assertCount(0, $response['registrationDetails']);
+        $this->assertNull($response['defaultMethod']);
+
+        /** @var MethodInterface $method */
+        $method = Injector::inst()->get(Method::class);
+        $loginHandler = $method->getLoginHandler();
+
+        $this->assertSame(
+            [$method->getURLSegment() => $loginHandler->getLeadInLabel()],
+            $response['registeredMethods']
+        );
+    }
+
+    public function testMFASchemaEndpointProvidesDefaultMethodIfSet()
+    {
+        // "Robbie" is security conscious and is also a CMS expert! He set up MFA and set a default method :o
+        $member = $this->objFromFixture(Member::class, 'robbie');
+        $this->scaffoldPartialLogin($member);
+
+        $result = $this->get('Security/login/default/mfa/schema');
+
+        $response = json_decode($result->getBody(), true);
+
+        $this->assertArrayHasKey('registeredMethods', $response);
+        $this->assertArrayHasKey('registrationDetails', $response);
+        $this->assertArrayHasKey('defaultMethod', $response);
+
+        $this->assertCount(1, $response['registeredMethods']);
+        $this->assertCount(0, $response['registrationDetails']);
+
+        /** @var RegisteredMethod $mathMethod */
+        $mathMethod = $this->objFromFixture(RegisteredMethod::class, 'robbie-math');
+        $this->assertSame($mathMethod->getMethod()->getURLSegment(), $response['defaultMethod']);
+    }
+
+    /**
+     * Mark the given user as partially logged in - ie. they've entered their email/password and are currently going
+     * through the MFA process
+     * @param Member $member
+     */
+    protected function scaffoldPartialLogin(Member $member)
+    {
+        $this->logOut();
+
+        $this->session()->set(SessionStore::SESSION_KEY, [
+            'member' => $member->ID,
+            'method' => null,
+            'state' => [],
+        ]);
+    }
+
+    /**
+     * @param Member $member
+     * @param string $password
+     * @return HTTPResponse
+     */
+    protected function doLogin(Member $member, $password)
+    {
+        $this->get(Config::inst()->get(Security::class, 'login_url'));
+
+        return $this->submitForm(
+            "MemberLoginForm_LoginForm",
```

Comment:
Minor note that the quotes here are different to the rest of the code

Reply:
Copy/paste from `SecurityTest.php` 🙁